### DISCUSSION TITLE
Imported A Peloton Ride Done Early May, No Distance

### DISCUSSION THREAD

**Discussion by saadmufti** [philosowaffle/peloton-to-garmin#167] [Q&A]
I followed all the steps on my MacBook, it all went fine and imported the one ride I did in early May. I am only interested in downloading the .FIT while and NOT uploading to Garmin, so it did that fine. But when I uploaded the .FIT file to Strava, it has no distance or speed. It does have power, heart rate, and cadence. The ride was a simulated ride in some Hawaii location. Any ideas why it wouldn't have distance or speed? The ride was on a Peloton Bike+, Thx

**Comment by philosowaffle** [comment]
Nothing initial comes to mind. To investigate further I would probably need you to share the original JSON workout with me([set json to True here](https://github.com/philosowaffle/peloton-to-garmin/wiki/Configuration#format-config)).

Also, I just want to note that if uploading to Strava is all you care about, Peloton already [natively supports syncing to strava](https://support.strava.com/hc/en-us/articles/115001304004-Peloton-and-Strava).

**Comment by saadmufti** [comment]
Hi, I'll try to grab a workout with json set to true.

As to why I'm doing this just to upload to Strava, there's a history there. I once created a Peloton profile on my bike and linked it to Strava, then didn't use it for a while and forgot the password. Long story short, I just deleted it and created a new profile, but now when I try to to link that new profile to Strava, the Peloton bike always gives me the error "this Strava account is already linked to a Peloton user". I have tried delinking from the Strava side, opening a ticket with Peloton, anything else I could think of, but Peloton support seems clueless. And I could find no way to delete the old profile even after recovering it. Even tried factory reset of the bike.

Thanks.

**Reply by philosowaffle** [reply]
Gotcha, that makes sense. I have never tested the files generated by P2G with Strava before, so we're in new territory here.  In theory, they should just work, but there's always a chance that Strava has some nuances I haven't accounted for.
